scipy/stats/_continuous_distns.py

```diff
@@ -2900,6 +2900,129 @@ def _entropy(self, c):
 genhalflogistic = genhalflogistic_gen(a=0.0, name='genhalflogistic')
 
 
+class genhyperbolic_gen(rv_continuous):
+    r"""A generalized hyperbolic continuous random variable.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The probability density function for `genhyperbolic` is:
+
+    .. math::
+
+        f(x, \lambda, \alpha, \beta, \delta, \mu) = 
+           \frac{(\gamma/\delta)^\lambda}{\sqrt{2\pi}K_\lambda(\delta \gamma)}
+           e^{\beta (x - \mu)} \times \frac{K_{\lambda - 1/2}
+           (\alpha \sqrt{\delta^2 + (x - \mu)^2})}
+           {(\sqrt{\delta^2 + (x - \mu)^2} / \alpha)^{1/2 - \lambda}}
+
+    for :math:`x \in (-\infty; \infty)`,
+    and :math:`\lambda, \mu \in \mathbb{R}`
+    ans :math: `0 \le |\beta| < \alpha ` \beta,
+    and :math: `\delta \in \mathbb{R}_{>0}`.
+    
+    Where: :math:`\gamma := \sqrt{\alpha^2-\beta^2}`.
+    :math: `K_{\lambda}(.)` denotes the modified Bessel function of the third
+    kind and order :math: `\lambda` (`scipy.special.kn`)
+
+    `genhyperbolic` takes ``lmbda`` as a tail parameter for :math:`\lambda`.
+    takes ``alpha`` as a shape parameter for :math:`\alpha`.
+    takes ``beta`` as a skeweness parameter for :math:`\beta`.
+    takes ``delta`` as a scale parameter for :math:`\delta`.
+    takes ``mu`` as a location parameter for :math:`\mu`.
```

Comment:
```suggestion
    `genhyperbolic` takes ``lmbda`` as a tail parameter for :math:`\lambda`,
    ``alpha`` as a shape parameter for :math:`\alpha`,
    ``beta`` as a skewness parameter for :math:`\beta`,
    ``delta`` as a scale parameter for :math:`\delta`,
    ``mu`` as a location parameter for :math:`\mu`.
```
Note that all SciPy distributions get `loc` and `scale` as subclasses of `rv_continuous`, so this might be over-parameterized.

Reply:
yes, I think one can set `delta = 1` in the standard form and `scale` takes care of the scaling. I would also suggest to write a and b in the methods of the class. It is easier to read and other distributions like `beta` use a and b instead of alpha and beta.

Reply:
@chrisb83 It is clear that `mu` is the same as `loc`, but looking closer, it's not obvious to me that there is a simple relationship between `delta` and `scale`. I can only see as far as factoring out `delta` from the terms:
![image](https://user-images.githubusercontent.com/6570539/103237131-98a6ee00-48fb-11eb-9332-e634c47986ec.png)

but that still leaves `delta` in the argument of the Bessel function and out of the argument of the exponential. 

Is there a simple relationship? If we can't provide the relationship between `delta` and `scale`, I imagine there will be a lot of issues raised about the parameterization.

Reply:
@mdhaber's concern is correct; with the current definition of the parameters, delta is not a true scale parameter.

We can parameterize the distribution as a true location-scale family if we modify the definitions of alpha and beta slightly.  Define

    alpha_hat = delta * alpha
    beta_hat = delta * beta

Then the parameters lambda, alpha_hat, beta_hat, mu, and delta give a true location-scale parameterization with location mu and scale delta.

I've written notes about it here: https://warrenweckesser.github.io/notes/genhypdist.html.  In those notes, (1) is the expression from wikipedia, but without the extra symbol gamma.  In (2), I made a few simply changes so that x always occurs in the form (x - mu)/delta.  The step from (2) to (3) is more algebra--tedious but straightforward.

Reply:
What this means is that we can remove mu and delta from the explicit set of shape parameters--the `rv_continuous` class will include `loc` and `scale` automatically.  That is, remove the method parameters `mu` and `delta`, and replace the values in the code with 0 and 1, respectively.  The docstring should note the difference in the meaning of the two shape parameters alpha and beta.  If we adopt @chrisb's suggestion to use `a` and `b` instead of `alpha` and `beta`, then we can use `alpha` and `beta` (or even α and β) for the parameterization used in the wikipedia article (and elsewhere), and `a` and `b` for the SciPy parameterization.  The docstring should explain the conversion, e.g. to go from wikipedia to SciPy, use

    lmbda = λ
    a     = δα 
    b     = δβ
    loc   = μ
    scale = δ

and to go from SciPy to wikipedia:

    λ = lmbda
    α = a/scale 
    β = b/scale
    μ = loc
    δ = scale

Reply:
Thank you all for the suggestion on the parameterization guys. @mdhaber yes `mu = loc` and the trick with `delta` is particularly elegant, @WarrenWeckesser . I will try and embed those change.

@chrisb83 I understand that `a` and `b` are in principle used to define the extremities of the pdf support [see here](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.rv_continuous.html). It is also true that some distributions just use those as parameters (see [norminversegauss](https://docs.scipy.org/doc/scipy/reference/tutorial/stats/continuous_norminvgauss.html)).  For consistency with related distributions I think `a` and `b` are anyways good candidates. Perhaps can you suggest is a better substitute for `lmbda` as well? I would want to avoid to have mixture of greek and latin letters.

Reply:
Looks like this has been implemented.